From 96fe63a2f5473f7ca686fd5ead937412d7084718 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20L=C3=B6ffler?= <st.loeffler@gmail.com>
Date: Thu, 11 Feb 2016 23:13:01 -0500
Subject: [PATCH 1/2] Fix bogus memory allocation in SplashFTFont::makeGlyph

---
 splash/SplashFTFont.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/splash/SplashFTFont.cc b/splash/SplashFTFont.cc
index 4cafbba4..37676a32 100644
--- a/splash/SplashFTFont.cc
+++ b/splash/SplashFTFont.cc
@@ -250,6 +250,9 @@ bool SplashFTFont::makeGlyph(int c, int xFrac, int yFrac, SplashGlyphBitmap *bit
     } else {
         rowSize = (bitmap->w + 7) >> 3;
     }
+    if (rowSize == 0 || bitmap->h == 0) {
+        return false;
+    }
     bitmap->data = (unsigned char *)gmallocn_checkoverflow(rowSize, bitmap->h);
     if (!bitmap->data) {
         return false;
-- 
2.25.1

